corehq/pillows/cacheinvalidate.py

```diff
@@ -2,84 +2,98 @@
-from couchforms.models import XFormInstance
-from pillowtop.listener import BasicPillow
+from pillowtop.checkpoints.manager import PillowCheckpoint
+from pillowtop.feed.couch import CouchChangeFeed
 from pillowtop.models import DjangoPillowCheckpoint
-
+from pillowtop.pillow.interface import ConstructedPillow
+from pillowtop.processors import PillowProcessor
 
 pillow_logging = logging.getLogger("pillowtop")
 
 
-class CacheInvalidatePillow(BasicPillow):
-    """
-    Pillow that listens to non xform/case _changes and invalidates the cache whether it's a
-    a single doc being cached, to a view.
-    """
+class FakeCheckpoint(PillowCheckpoint):
+    def __init__(self, checkpoint_id, couch_db):
+        super(FakeCheckpoint, self).__init__("{}_{}".format(checkpoint_id, couch_db.dbname))
+        self.couch_db = couch_db
 
-    def __init__(self, pillow_id, couch_db, couch_filter=None):
-        self._pillow_id = pillow_id
-        self.couch_filter = couch_filter
-        super(CacheInvalidatePillow, self).__init__(couch_db=couch_db)
-        self.gen_caches = set(GenerationCache.doc_type_generation_map().values())
+    def get_or_create_wrapped(self, verify_unchanged=False):
+        return DjangoPillowCheckpoint(
+            checkpoint_id=self.checkpoint_id,
+            sequence=self.couch_db.info()['update_seq'],
+        )
 
-    @property
-    def pillow_id(self):
-        return self._pillow_id
+    def get_current_sequence_id(self):
+        return self.couch_db.info()['update_seq']
 
-    def set_checkpoint(self, change):
-        """
-        Override to do nothing - don't want to cause doc update conflicts with invalidation
-        """
+    def update_to(self, seq):
         pass
 
-    def reset_checkpoint(self):
+    def reset(self):
+        pass
+
+    def touch(self, min_interval):
         pass
 
-    def get_checkpoint(self, verify_unchanged=False):
-        return DjangoPillowCheckpoint(
-            checkpoint_id=self.checkpoint.checkpoint_id,
-            sequence=self.get_couch_db().info()['update_seq'],
-        )
+
+class CacheInvalidateProcessor(PillowProcessor):
+    def __init__(self):
+        self.gen_caches = set(GenerationCache.doc_type_generation_map().values())
 
     def get_generations(self):
         return ["%s :: %s" % (gc.generation_key, gc._get_generation()) for gc in self.gen_caches]
 
-    def change_trigger(self, changes_dict):
+    def process_change(self, pillow_instance, change):
+        self.process_doc(change.get_document(), change.deleted)
+
+    def process_doc(self, doc, is_deleted):
         """
         This function does actual cache invalidation. It's also called manually
         by directly invalidated things.
         """
-        # requires pillowtop to listen with include_docs
-        doc = changes_dict['doc']
         doc_id = doc['_id']
-        deleted = changes_dict.get('deleted', False)
-
         if doc_id.startswith('pillowtop_corehq.pillows'):
```

Comment:
not related to your changes, but I think we can rid of this since we moved checkpoints to sql